crates/ark/src/data_explorer/convert_to_code.rs

```diff
@@ -0,0 +1,854 @@
+            ResolvedSortKey {
+                column_name: "sales".to_string(),
+                ascending: false,
+            },
+            ResolvedSortKey {
+                column_name: "region".to_string(),
+                ascending: true,
+            },
+        ];
+
+        let result = convert_to_code(params, Some("my_data"), &resolved_sorts);
+
+        assert_eq!(result.converted_code, vec![
+            "library(dplyr)".to_string(),
+            "".to_string(),
+            "my_data |>\n  filter(\n    price > 100,\n    category == \"Electronics\"\n  ) |>\n  arrange(desc(sales), region)".to_string(),
+        ]);
+    }
+
+    #[test]
+    fn test_default_object_name() {
+        let params = ConvertToCodeParams {
+            column_filters: vec![],
+            row_filters: vec![],
+            sort_keys: vec![],
+            code_syntax_name: amalthea::comm::data_explorer_comm::CodeSyntaxName {
+                code_syntax_name: "dplyr".to_string(),
+            },
+        };
+
+        let result = convert_to_code(params, None, &[]);
+
+        assert_eq!(result.converted_code, vec![
+            "library(dplyr)".to_string(),
+            "".to_string(),
+            "dat".to_string(),
+        ]);
+    }
+}
+
+/// Suggest a code syntax based on available options
+///
+/// Currently always returns "dplyr" as the preferred syntax
+///
+/// # Returns
+///
+/// A `CodeSyntaxName` with the suggested syntax
+pub fn suggest_code_syntax() -> CodeSyntaxName {
+    CodeSyntaxName {
+        code_syntax_name: "dplyr".into(),
+    }
+}
+
+#[cfg(test)]
+mod execution_tests {
+    use super::*;
+    use crate::fixtures::r_test_lock;
+    use crate::r_task::r_task;
+    use amalthea::comm::data_explorer_comm::CodeSyntaxName;
+    use amalthea::comm::data_explorer_comm::ColumnDisplayType;
+    use amalthea::comm::data_explorer_comm::ColumnSchema;
+    use amalthea::comm::data_explorer_comm::ConvertToCodeParams;
+    use amalthea::comm::data_explorer_comm::FilterComparison;
+    use amalthea::comm::data_explorer_comm::FilterComparisonOp;
+    use amalthea::comm::data_explorer_comm::RowFilter;
+    use amalthea::comm::data_explorer_comm::RowFilterCondition;
+    use amalthea::comm::data_explorer_comm::RowFilterParams;
+    use amalthea::comm::data_explorer_comm::RowFilterType;
+    use harp::DataFrame;
+
+    /// Helper function to execute generated convert-to-code output and assign the
+    /// result
+    fn execute_generated_code_and_assign_result(
+        mut code_lines: Vec<String>,
+        result_name: &str,
+    ) -> anyhow::Result<()> {
+        r_task(|| -> anyhow::Result<()> {
+            let last_line = code_lines.pop().unwrap();
+            code_lines.push(format!("{} <- {}", result_name, last_line));
+
+            let full_code = code_lines.join("\n");
+            harp::parse_eval_global(&full_code)?;
+
+            Ok(())
+        })
+    }
+
+    #[test]
+    fn test_convert_to_code_execution_basic_filter() {
```

Comment:
I don't think this test is terribly interesting as it is. I consider more of an MVP for a type of test where we actually execute the generated code and assert things about the result. I could imagine this being a useful tool as we build this feature out. Similar tests are used on the Python side.